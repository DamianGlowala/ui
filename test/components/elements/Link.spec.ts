import { describe, it, expect } from 'vitest'
import { ULink } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Link', () => {
  it.each([
    ['basic case', {}],
    ['with custom as prop', { props: { as: 'a' } }],
    ['with custom type prop', { props: { type: 'submit' } }],
    ['disabled set to true', { props: { disabled: true } }],
    ['active set to true', { props: { active: true } }],
    ['exact set to true', { props: { exact: true } }],
    ['exactQuery set to true', { props: { exactQuery: true } }],
    ['exactHash set to true', { props: { exactHash: true } }],
    ['with inactiveClass prop', { props: { inactiveClass: 'inactive-link' } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Link.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, ULink)
    expect(html).toMatchSnapshot()
  })
})
