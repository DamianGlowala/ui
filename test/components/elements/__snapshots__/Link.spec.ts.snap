// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > renders NuxtLink with slot correctly 1`] = `"<a>nuxtlink slot</a>"`;

exports[`Link > renders NuxtLink with to prop correctly 1`] = `"<a></a>"`;

exports[`Link > renders basic case correctly 1`] = `"<button type="button"></button>"`;

exports[`Link > renders disabled set to true correctly 1`] = `"<button type="button" disabled=""></button>"`;

exports[`Link > renders with <component as slot correctly 1`] = `"<a type="button">component slot</a>"`;

exports[`Link > renders with active correctly 1`] = `"<a class="text-sm"></a>"`;

exports[`Link > renders with custom as prop correctly 1`] = `"<a type="button"></a>"`;

exports[`Link > renders with custom type prop correctly 1`] = `"<button type="submit"></button>"`;

exports[`Link > renders with inactive correctly 1`] = `"<a class="text-gray-300"></a>"`;
